service/src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -206,16 +206,34 @@ public ApiJobReport mapFlightStateToApiJobReport(FlightState flightState) {
     String completedDate = null;
     HttpStatus statusCode = HttpStatus.ACCEPTED;
 
-    if (flightState.getCompleted().isPresent()) {
-      FlightMap resultMap = getResultMap(flightState);
-      // The STATUS_CODE return only needs to be used to return alternate success responses.
-      // If it is not present, then we set it to the default OK status.
-      statusCode = resultMap.get(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.class);
-      if (statusCode == null) {
-        statusCode = HttpStatus.OK;
-      }
-
+    if (jobStatus != StatusEnum.RUNNING) {
+      // If the job is completed, the JobReport should include a result code indicating success or
+      // failure. For failed jobs, this code is the error code. For successful jobs, this is the
+      // code specified by the flight if present, or a default of 200 if not.
       completedDate = flightState.getCompleted().get().toString();
+      switch (jobStatus) {
+        case FAILED:
+          int errorCode =
+              flightState
+                  .getException()
+                  .map(e -> ErrorReportUtils.buildApiErrorReport(e).getStatusCode())
+                  .orElseThrow(
+                      () ->
+                          new InvalidResultStateException(
+                              "Flight failed with no exception reported"));
+          statusCode = HttpStatus.valueOf(errorCode);
+          break;
+        case SUCCEEDED:
+          FlightMap resultMap = getResultMap(flightState);
+          statusCode = resultMap.get(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.class);
+          if (statusCode == null) {
+            statusCode = HttpStatus.OK;
```

Comment:
I wonder if this isn't masking a potential problem. Should we warn in this case or something? We throw for unknown status but not for unknown HttpStatus?

Reply:
I don't think this is a problem - not all flights explicitly set a response status, mostly because we've been defaulting to 200 already. So this case is more "default HttpStatus" than "unknown HttpStatus".

I think throwing on the unknown jobStatus is just best practice for enums in switch statements - we've already made explicit cases for all current enum values of jobStatus, so I don't expect to ever hit this case (unless someone adds new jobStatus values incorrectly, in which case this will catch that error).